src/withAxiom.ts

```diff
@@ -122,10 +123,15 @@ function isNextConfig(param: WithAxiomParam): param is NextConfig {
 }
 
 function isApiHandler(param: WithAxiomParam): param is NextApiHandler {
-  // This is pretty hacky, but if you call withAxiom in a serverless function,
-  // the environment variable will be set.
-  // The middleware runs on CloudFlare workers which doesn't expose that env.
-  return typeof param == 'function' && !!process.env.LAMBDA_TASK_ROOT;
+  const isFunction = typeof param == 'function';
+
+  // check if running locally
+  if (process.env.NODE_ENV === 'development') {
+    const isLocalWorker = caller() == 'evalmachine.<anonymous>';
+    return isFunction && !isLocalWorker;
+  }
+  const isLambda = !!process.env.LAMBDA_TASK_ROOT;
+  return isFunction && isLambda;
```

Comment:
Maybe we can return early if it's not a function? Might make things easier to read 🤔

Reply:
updated

Reply:
We don't need `isFunction` in the return statements anymore, it's always true